### PR TITLE
Use install-dotnet scripts for installation

### PR DESCRIPTION
**Pull request by ZEisinger**
Fixes sxs issues (#25), caching, allows using global.json (#14), better error messages (#54)

Default install location is used rather than the toolkit's cache, this enables side by side installs to work properly.

### CODE REVIEW

File: src/installer.ts [hunk 1]
```diff
@@ -294,84 +61,40 @@ export class DotnetCoreInstaller {
...
 
-    let primaryUrl: string = '';
-    let legacyUrl: string = '';
-    if (!!output && output.length > 0) {
-      let lines: string[] = output.split(os.EOL);
-
-      // Fallback to \n if initial split doesn't work (not consistent across versions)
-      if (lines.length === 1) {
-        lines = output.split('\n');
+      let scriptArguments: string[] = [];
```
Comment (ZEisinger) [inline src/installer.ts]: Need to pass proxy settings into the script when being called

File: src/installer.ts [hunk 2]
```diff
@@ -1,287 +1,54 @@
...
-
-    let resultCode = 0;
     if (IS_WINDOWS) {
       let escapedScript = path
-        .join(__dirname, '..', 'externals', 'get-os-platform.ps1')
+        .join(__dirname, '..', 'externals', 'install-dotnet.ps1')
         .replace(/'/g, "''");
       let command = `& '${escapedScript}'`;
+      if (this.version) {
+        command += ` -Version ${this.version}`;
```
Comment (ZEisinger) [inline src/installer.ts]: Need to pass in proxy settings

**Comment by RehanSaeed** [comment]
Any updates on this?

**Comment by int3rrupt** [comment]
@RehanSaeed I’ve been using for a while now. I know it’s not the best solution but you’re able to use this by referencing the branch name whole this gets merged.
```
- name: Setup .NET Core 2.1
      uses: actions/setup-dotnet@sxs-with-fallback
      with:
        dotnet-version: 2.1.804
```

**Comment by ZEisinger** [comment]
The install scripts are incompatible with proxy settings added recently. The product team has also pushed back on autoupgrade scenarios being added. Addressing #25 will need to be done separately.

**Comment by RehanSaeed** [comment]
Is this now released?

**Comment by int3rrupt** [comment]
> Is this now released?

@RehanSaeed Looks like a new PR was created for side by side installs #124

**Comment by ZEisinger** [comment]
There were a lot of changes since this PR, so I created a clean PR

**Pull request closed without merging**